src/decoder/ifd.rs

```diff
@@ -332,171 +332,191 @@ impl Entry {
         limits: &super::Limits,
         decoder: &mut super::Decoder<R>,
     ) -> TiffResult<Value> {
+        // Case 1: there are no values so we can return immediately.
+        if self.count == 0 {
+            return Ok(List(Vec::new()));
```

Comment:
Not sure if an empty list is the right thing to return, but that's what the code was doing before.